### ISSUE TITLE
@aws-cdk/app-staging-synthesizer-alpha: Uses default bootstrap exec role when bootstrapQualifier is supplied

### ISSUE THREAD

**Issue by BrianFarnhill** [aws/aws-cdk#28195]
### Describe the bug

Consider the following use of the `AppStagingSynthesizer`:

```
defaultStackSynthesizer: AppStagingSynthesizer.defaultResources({
    appId: 'UniqueAppId',
    bootstrapQualifier: 'Custom',
    deploymentIdentities: DeploymentIdentities.defaultBootstrapRoles({
      bootstrapRegion: 'ap-southeast-2',
    }),
  }),
```

In this scenario I'm trying to avoid the use of the default qualifier at all (to meet a customer requirement, each bootstrap uses a unique qualifier that enforces a specific permission boundary for each app that is deployed). When I deploy this the app works and my main CDK stacks all use the CFN exec role that goes with the 'Custom' qualifier in this case, but the Staging Stack itself still relies on the exec role from the default bootstrap. I would expect that given I've specified a qualifier here that it would use that for the staging stack as well, but if that assumption is wrong, having an option to specific the exec role for the staging stack would also address the concern. 

### Expected Behavior

I would expect the deployment role used on the staging stack to match that of the `Custom` bootstrap qualifier that was deployed in to the account.

### Current Behavior

It deploys with the default bootstrap qualifier instead (being `hnb659fds` and the role `arn:aws:iam::{ACCOUNT}:role/cdk-hnb659fds-cfn-exec-role-{ACCOUNT}-{REGION}` if you don't customise the bootstrap)

### Reproduction Steps

Use the code snippet from the example - you'll need to create a secondary bootstrap qualifier as well as the default one though, so the flow would be:

1. Deploy the default bootstrap with `cdk bootstrap aws://ACCOUNT-NUMBER-1/REGION-1` (using your account number and region)
2. Deploy a custom bootstrap with `cdk bootstrap --qualifier Custom aws://ACCOUNT-NUMBER-1/REGION-1` (using your account number and region)
3. Use the snippet above with the qualifier name and appropriate region name
4. Deploy any CDK stack against that synthesiser

Go in here and look at the deployed stack, you'll see the StagingStack uses the default qualifier. You'll also see that role called out in the `manifest.json` file in the `cdk.out` directory for the staging stack:

```
"cloudFormationExecutionRoleArn": "arn:${AWS::Partition}:iam::${AWS::AccountId}:role/cdk-hnb659fds-cfn-exec-role-${AWS::AccountId}-${AWS::Region}",
```

### CDK CLI Version

2.100.0 (build e1b5c77)

### Node.js Version

v18.18.0

### OS

Linux and MacOs

### Language

TypeScript

### Language Version

TypeScript (5.5.2)

**Comment by pahud** [comment]
Yes I can reproduce this, it's still deploying with the role using `hnb659fds` qualifier.

**Comment by sakurai-ryo** [comment]
We are using `BootstraplessSynthesizer` to deploy the `StagingStack`, but we can't pass `qualifier` to it like we can to `DefaultStackSynthesizer `.
So need to update the `BootstraplessSynthesizerProps` to accept arguments such as `qualifier`.
https://github.com/aws/aws-cdk/blob/358ceadd3352b4c692438b9d9061354556fc5bac/packages/%40aws-cdk/app-staging-synthesizer-alpha/lib/default-staging-stack.ts#L262

I'll submit a PR when I have time.

**Comment by ali-csiro** [comment]
Was wondering if anyone had found a workaround to this please, or if there is any updates to this issue? We're using a bootstrap situation that is identical to the one @BrianFarnhill describes above, and we need cross account deployments from code pipeline. I'd like to avoid going back to default stack synthesizer, but I cannot find a way around the issue!

**Comment by pahud** [comment]
### Expanded Scope: DeploymentIdentities Also Affected

Testing shows this bug affects both configuration methods:

1. ✅ Originally reported: bootstrapQualifier ignored
2. ✅ Newly confirmed: DeploymentIdentities.specifyRoles() also ignored

Test case 
```typescript
const app = new cdk.App({
  defaultStackSynthesizer: AppStagingSynthesizer.defaultResources({
    appId: 'my-app-id',
    stagingBucketEncryption: BucketEncryption.S3_MANAGED,
    bootstrapQualifier: "test-custom-name",
    deploymentIdentities: DeploymentIdentities.specifyRoles({
      cloudFormationExecutionRole: BootstrapRole.fromRoleArn('arn:aws:iam::TargetAccount:role/Execute'),
      deploymentRole: BootstrapRole.fromRoleArn('arn:aws:iam::TargetAccount:role/Deploy'),
      lookupRole: BootstrapRole.fromRoleArn('arn:aws:iam::TargetAccount:role/Lookup'),
    })
  })
});
```


Result: Still attempts to use cdk-hnb659fds-deploy-role-ACCOUNT-REGION instead of the specified custom roles.

### Root Cause Analysis

The issue is a disconnect between policy generation and deployment execution:

• ✅ Policy Generation: Custom roles are correctly used in IAM and S3 bucket policies within the Staging Stack template
• ❌ Deployment Execution: The BootstraplessSynthesizer ignores custom configuration and falls back to default roles

Technical Root Cause: In [default-staging-stack.ts line 262](https://github.com/aws/aws-cdk/blob/main/packages/@aws-cdk/app-staging-synthesizer-alpha/lib/default-staging-stack.ts#L262):

```typescript
super(scope, id, {
  ...props,
  synthesizer: new BootstraplessSynthesizer(), // ❌ No custom config passed
  // ...
});
```


The props.deployRoleArn and props.qualifier are available but not forwarded to the synthesizer.

### Proposed Fix

The fix should pass the deployment configuration to BootstraplessSynthesizer:

```typescript
synthesizer: new BootstraplessSynthesizer({
  deployRoleArn: props.deployRoleArn,
  cloudFormationExecutionRoleArn: context.cloudFormationExecutionRoleArn,
}),
```


This would require extending `BootstraplessSynthesizerProps` to accept these parameters.

### Impact

This bug breaks core functionality of the app-staging-synthesizer-alpha module for:
• Cross-account deployments with custom qualifiers
• Organizations requiring custom bootstrap roles for security/compliance
• Any scenario where default bootstrap roles don't exist
